client/src/pages/reports/Form.js

```diff
@@ -941,13 +948,21 @@ const ReportForm = ({
                         name="reportSensitiveInformation.text"
                         component={FieldHelper.SpecialField}
                         label="Report sensitive information text"
-                        onChange={value =>
-                          setFieldValue(
-                            "reportSensitiveInformation.text",
-                            value || null,
-                            true
-                          )
-                        }
+                        onChange={value => {
+                          const safeVal = value || null
+                          if (
+                            !_isEqual(
+                              values.reportSensitiveInformation.text,
+                              safeVal
+                            )
+                          ) {
+                            setFieldValue(
+                              "reportSensitiveInformation.text",
+                              value,
```

Comment:
Yes it should be safeVal, btw the report sensitive info always being set to { uuid: null, text: null} whenever autosave runs because of these queries :
```
const GQL_CREATE_REPORT = gql`
  mutation($report: ReportInput!) {
    createReport(report: $report) {
      uuid
      state
      authors {
        uuid
      }
      reportSensitiveInformation {
        uuid
        text
      }
    }
  }
`
const GQL_UPDATE_REPORT = gql`
  mutation($report: ReportInput!, $sendEditEmail: Boolean!) {
    updateReport(report: $report, sendEditEmail: $sendEditEmail) {
      uuid
      state
      authors {
        uuid
      }
      reportSensitiveInformation {
        uuid
        text
      }
    }
  }
`
```

Reply:
So anyone with an edit access to the report can already get a `reportSensitiveInformation` to be `{ uuid: null, text: null }`

Reply:
Looking at the server-side code, there are three possibilities:

1. user has no access to the rsi: `null` is returned
1. user has access to the rsi, but the report has no rsi: 'empty' rsi is returned (`{uuid: null, text: null}`)
1. user has access to the rsi, and the report has rsi: the rsi is returned (`{uuid: "…", text: "…"}`)

Reply:
Hmm, so i have this scenario (btw, we are talking about Form page, not Show page):
Scenario 1:
-  "Person1" creates a report, adds sensitive info authorized for "Organization1", also adds a task from "Organization2". Sends the report.
- Task owner "Person2" from "Organization2" will approve this report, so he has Edit access, "Person2" doesn't see the sensitive info, opens the form, **now can "Person2" add another sensitive info here ??**  
- Also, "Person2" edits the report, "autosave" runs after 30 sec, resulting in this query:
```
const GQL_UPDATE_REPORT = gql`
  mutation($report: ReportInput!, $sendEditEmail: Boolean!) {
    updateReport(report: $report, sendEditEmail: $sendEditEmail) {
      uuid
      state
      authors {
        uuid
      }
      reportSensitiveInformation {
        uuid
        text
      }
    }
  }
`
```
Now, if `null` returns from the server, and the report will be merged with updated one returning from query, **will the Form page "Person2" is looking have a `reportSensitiveInformation` set to `{uuid: null, text: null}` ? or `null` ?**

Reply:
> now can "Person2" add his own sensitive info here ??
Of course not. Someone without access to the rsi can neither read nor write it; the server-side forbids it.

Reply:
I am not talking about the existing sensitive info saved by "Person1", since "Person2" has an empty sensitive info when started editing the form, from his perspective, he can add a new one. So basically, a report can have more than one sensitive infos. 
- "SensitiveInfo1" for "Organization1" added by "Person1" when creating
- "SensitiveInfo2" for "Organization2" added by "Person2" while editing.

Reply:
> So basically, a report can have more than one sensitive infos.

No. There's even a UNIQUE INDEX on database level to prevent that.

Reply:
There is no check to prevent that on client side, that is why i was curios.

Reply:
Because this is always `true` since we initialize with `{}` or my proposal.
```
                <Collapse in={showSensitiveInfo}>
                  {(values.reportSensitiveInformation || !edit) && (...
```

Reply:
Now i get it, Collapse was erroring out if it was null, so someone set it to truthy value, but forgot about the implications. We actually want to it become null when an unauthorized person editing the page, shouldn't see the sensitive info component.